roblox/catalog.py

```diff
@@ -0,0 +1,108 @@
+"""
+
+This module contains classes intended to parse and deal with data from Roblox catalog endpoint.
+
+"""
+from __future__ import annotations
+from datetime import datetime
+
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from .client import Client
+    from typing import Optional
+from .bases.basecatalogitem import BaseCatalogItem
+from .bases.baseuser import BaseUser
+from .assets import AssetType
+from .partials.partialgroup import PartialGroup
+from .partials.partialuser import CatalogCreatorPartialUser
+
+class CatalogItem(BaseCatalogItem):
+    """
+    Represents a Catalog/Avatar Shop/Marketplace item.
+
+    Attributes:
+        id: The item's ID.
+        name: The item's name.
+        item_type: Unknown.
+        asset_type: The asset's type as an instance of AssetType
+        description: The item's description.
+        is_offsale: If the item is offsale.
+        creator: A class representing the creator of the item.
+        price: The price of the item, in Robux.
+        purchase_count: The number of times the item has been purchased.
+        favorite_count: The number of times the item has been favorited.
+        sale_location_type: Unknown.
+        premium_pricing: A dictionary storing information about pricing for Roblox Premium members.
+        premium_pricing.in_robux: The pricing for Roblox Premium members, in Robux.
+        premium_pricing.discount_percentage: The percentage that Roblox Premium members get discounted.
+    """
+
+    def __init__(self, client: Client, data: dict):
+        self._client: Client = client
+        self.id: int = data["id"]
+        self.item_type = data["itemType"]
+        super().__init__(client=self._client, catalog_item_id=self.id, catalog_item_type=self.item_type)
+
+        self.name: str = data["name"]
+        self.description: str = data["description"]
+
+        self.asset_type: AssetType = AssetType(type_id=data["assetType"])
+
+        self.is_offsale: bool = data["isOffsale"]
+
+        # Creator
+        self.creator: CatalogCreatorPartialUser or CatalogCreatorPartialGroup
+        if data["creatorType"] == "User":
+            self.creator = CatalogCreatorPartialUser(client=client, data=data)
+        elif data["creatorType"] == "Group":
+            self.creator = CatalogCreatorPartialGroup(client=client, group_id=data)
+
+        self.price: int = data["price"]
+        self.purchase_count: int = data["purchaseCount"]
+        self.favorite_count: int = data["favoriteCount"]
+        self.sale_location_type: str = data["saleLocationType"]
+
+
+
+        if data["premiumPricing"]:
+            self.premium_pricing = {}
+            self.premium_pricing.in_robux: int = data["premiumPricing"]["premiumPriceInRobux"]
```

Comment:
not how a dict works. also, why is this a dict and not a new class?

Reply:
:see_no_evil: Clearly I'm spending too much time in Lua-land /lh